packages.dhall

```diff
@@ -41,6 +41,11 @@
   , repo = "https://github.com/purenix-org/purescript-distributive.git"
   , version = "master"
   }
+, effect =
+  { dependencies = [ "prelude" ]
+  , repo = "https://github.com/purenix-org/purescript-effect.git"
+  , version = "main"
```

Comment:
Feel free to cut a version for this.  I've been meaning to do it with the other packages as well, but I've put it off so far.

Reply:
Yeah, that's definitely something we should start doing. But do we already have a versioning strategy for this? I mean we could do a completely independent semver, or we could try to reflect the relation to the PS/JS package we forked from.

So it could be either a `1.0.0` or something that would refer to the current `3.0.0` of the original package. Or a mix of both?

Reply:
Yeah, that's a good question.

I was thinking of mirroring the upstream versions (since all the current packages are direct forks of the upstream packages), but that brings up a couple questions:

- What if we have to do a patch release (or major or minor release) that upstream doesn't have?  Then we will forever be one version ahead.
- What if upstream does a release, but it only includes fixes to the underlying .js files?  We may not even need to do a release at all.
- What do we do for packages like `purescript-effect` where we may significantly deviate from upstream?

It may be best to just start out everything at 1.0.0.  Although it might become tricky to figure out what changes from the upstream packages have been applied to our `purenix-org/purescript-*` packages.

Reply:
how about this:

`1.0.0-based-on-3.0.0` (just an example syntax)

In the first version we just do our fully independent semver. The second version means something like: "The author that released this version tried to be as close as possible to this exact upstream version."

So the cases in your bullet points would be then:
- We would jump to `1.0.1-based-on-3.0.0`
- We would either just stay at `1.0.0-based-on-3.0.0`. It's ok to be a little bit behind. Or we'd do a dummy release of `1.0.0-based-on-3.0.1`
- That would be ok. "as close as possible" can mean matching the API 100%, or if not possible only 20%.

Just a spontaneous idea. Not sure, if it's practicable. As the porting is a manual process, I think there cannot be a fully formalized approach.

Reply:
Oh, that's an interesting idea.  I wonder if `spago` would accept a version like that?

Reply:
I'm pretty sure it would take every ["branch, tag, or commit hash"](https://github.com/purescript/spago#add-a-package-to-the-package-set).
Apart from that, the semver spec even allows something like "build metadata" that can be added to a version: https://semver.org/#spec-item-10